### DISCUSSION TITLE
How to make an Url optional?

### DISCUSSION THREAD

**Discussion by awhitford** [open-circle/valibot#485] [General]
I am running into a problem creating an **optional url** because an _empty string_ is not considered a valid url.  For example:
```
v.optional(v.string([v.maxLength(255), v.url()])) 
```
doesn't seem to accept an empty string (which would be a typical optional value on a web form).

**Comment by fabian-hiller** [comment]
I recommend to use `union`:

- https://valibot.dev/guides/unions/
- https://valibot.dev/api/union/

```ts
import * as v from 'valibot';

const UrlSchema = v.union([v.literal(''), v.string([v.maxLength(255), v.url()])]);
```
